src/Microsoft.ApplicationInsights/DataContracts/TelemetryContext.cs

```diff
@@ -156,13 +156,13 @@ internal void Initialize(TelemetryContext source, string instrumentationKey)
         {
             Property.Initialize(ref this.instrumentationKey, instrumentationKey);
 
-            this.component?.CopyFrom(source);
-            this.device?.CopyFrom(source);
-            this.cloud?.CopyFrom(source);
-            this.session?.CopyFrom(source);
-            this.user?.CopyFrom(source);
-            this.operation?.CopyFrom(source);
-            this.location?.CopyFrom(source);
+            this.Component.CopyFrom(source);
```

Comment:
This may be the simplest fix, but it's defeating the purpose of making the backing fields lazy.
Consider:
`if (source.component != null) this.Component.CopyFrom(source);`
etc.

Reply:
Isn't the entire check backwards? I mean..... whether the target is null or not seems (to me) irrelevant. Wouldn't it be better to check if the *source* has a property set that requires copying? And if so, the target being null is clearly not an issue since you're gonna have to create it in the first place. Am I misunderstanding the whole thing?

Reply:
@tomasr you're right.
There was some "magic" happening here to LazyInitialize fields as needed.
I've refactored all of this to make it more readable.